subsys/net/lib/openthread/platform/radio.c

```diff
@@ -102,6 +110,79 @@ void energy_detected(struct device *dev, s16_t max_ed)
 	}
 }
 
+/**
+ * @brief If radio signal is set, read it and clear
+ *
+ * @param signal_id     Identifier openthread signal to read
+ * @param signal_value  Value of obtained signal
+ *
+ * @retval              True if signal is valid
+ *
+ */
+static bool radio_signal_read(enum radio_signal_id signal_id,
+					int *signal_value)
+{
+	struct k_poll_signal *signal = &radio_signals[signal_id];
+	int set;
+	bool success = false;
+
+	k_poll_signal_check(signal, &set, signal_value);
+
+	if (set) {
+		k_poll_signal_reset(signal);
+		success = true;
+	}
+	return success;
+}
+
+/**
+ * @brief Sets radio signal with given value
+ *
+ * @param signal_id     Identifier of radio signal
+ * @param signal_value  Value of signal to set
+ *
+ */
+static void radio_signal_set(enum radio_signal_id signal_id,
+					int signal_value)
+{
+	struct k_poll_signal *signal = &radio_signals[signal_id];
+
+	k_poll_signal_raise(signal, signal_value);
+}
+
+/**
+ * @brief Wait for specified radio signals.
+ *
+ * @param signal_start_id  Identifier of first radio signal
+ * @param num_events       Number of signals
+ * @param timeout          Timeut in us
+ *
+ * @retval 0 One or more signals are ready.
+ * @retval -EAGAIN Waiting period timed out.
+ * @retval other -- see k_poll system call.
+ */
+static int radio_signal_poll(enum radio_signal_id signal_start_id,
+					int num_events, s32_t timeout)
+{
+	return k_poll(&radio_events[signal_start_id], num_events,
+							timeout);
+}
+
+/**
+ * @brief Inits radio signals
+ *
+ */
+static void radio_signal_init(void)
+{
+	for (int i = 0; i < RADIO_SIGNALS; i++) {
```

Comment:
Are you planning to introduce more of these signals in the future? It's introducing **a lot** of new code to add functionality that could be covered with a single semaphore. For the ED result, it would even make sense to reuse `energy_detected_value` global variable.

Reply:
I agree with Robert. we have no plans for introducing more signals and current solution requires additional semaphore to be truly thread safe.
I would suggest making solution with semaphore + reuse of energy_detected_value (but do not reuse pending event bit as this may introduce issues in the future when threading configuration is changed). If you prefer using signals you could merge poll and read function into otPlatRadioGetRssi and use INITIALIZER macro instead of init functions: K_POLL_SIGNAL_INITIALIZER(obj)
K_POLL_EVENT_INITIALIZER(event_type, event_mode, event_obj)
K_POLL_EVENT_STATIC_INITIALIZER(event_type, event_mode, event_obj, event_tag) 
This I think will reduce a lot of overhead code.
If in the future we will need to make a general api here we can easily recreate it.